gems/pending/openstack/openstack_handle/handled_list.rb

```diff
@@ -0,0 +1,118 @@
+module OpenstackHandle
+  module HandledList
+    def handled_list(collection_type, options = {})
+      # Alias with hardcoded :all method, just for convenience, since :all method will be used in 99% of the cases
+      handled_list_method(collection_type, options, :all)
+    end
+
+    def handled_list_method(collection_type, options = {}, method = :all)
+      # Will automatically handle multi-tenancy and pagination of all Fog list methods, so we always get all openstack
+      # entities back. The exceptions of each service and collection type will be solved in <service_name>)_delegate
+      # classes by defining multi_tenancy_type and pagination_type methods base on collection_type
+      # Example of call in refresh code
+      # @compute_service.handled_list(:servers)
+      # @orchestration_service.handled_list(:resources, :stack => stack)
+      # By default, it calls :all method on the Fog collection, that has unified interface in all list methods in Fog
+      # and always returns detailed list.
+      case multi_tenancy_type(collection_type)
+      when :all_tenants_loop
+        paginated_list_method = -> (x) { x.get_paginated_list(collection_type, options, method) }
```

Comment:
I think rubocop says stabby for single-line... but that's from memory.

I'm pro-stabby, but I'm also pro-new-hash (not because I specifically love them, but because they're Normal Modern Ruby As Defined By Matz)